velox/expression/RowConstructor.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/RowConstructor.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::exec {
+
+const char* const kRowConstructor = "row_constructor";
+
+TypePtr RowConstructorCallToSpecialForm::resolveType(
+    const std::vector<TypePtr>& argTypes) {
+  auto numInput = argTypes.size();
+  std::vector<std::string> names(numInput);
+  std::vector<TypePtr> types(numInput);
+  for (auto i = 0; i < numInput; i++) {
+    types[i] = argTypes[i];
+    names[i] = fmt::format("c{}", i + 1);
+  }
+  return ROW(std::move(names), std::move(types));
+}
+
+ExprPtr RowConstructorCallToSpecialForm::constructSpecialForm(
+    const std::string& name,
+    const TypePtr& type,
+    std::vector<ExprPtr>&& compiledChildren,
+    bool trackCpuUsage,
+    const core::QueryConfig& config) {
+  static auto rowConstructorVectorFunction =
```

Comment:
should this be static given it depends on config and name?

one could register RowConstructor with two different names, or two queries could run with two different configs

I know it was static before, but the name was fixed, and I'm not sure if it was correct, but it probably didn't matter since RowConstructor didn't actually depend on config

Reply:
SpecialForm does not support prefix, it has one name.
But it may run with tow different configs, so I will change it to non static.